src/test/java/com/mongodb/hibernate/jdbc/MongoConnectionTests.java

```diff
@@ -63,6 +68,10 @@ class MongoConnectionTests {
     @Mock
     private MongoClient mongoClient;
 
+    @Spy
+    private MongoDialectSettings config =
+            MongoDialectSettings.builder(emptyMap()).databaseName("testDbName").build();
```

Comment:
I had to use `@Spy` instead of `@Mock` here so that Mockito injected the value of the `config` field specified in the code into `mongoConnection`, as opposed to generating the `config` value on its own.

Reply:
This approach works. More intuitive and idematic way might be:

```
    @Mock(lenient = true)
    private MongoDialectSettings config;

    @InjectMocks
    private MongoConnection mongoConnection;

    @BeforeEach
    void setUp() {
        doReturn("testDbName").when(config).getDatabaseName();
    }
```
the benefit is subjective. It ends up with avoiding an unnecessary hack so it is easy to understand and change in the long future.

Reply:
Agreed to create `MongoConnection` manually and get rid of `@InjectMocks`. That way we don't need to `@Mock`/`@Spy` on `MongoDialectSettings`.

Reply:
Done in fdc4c5feb1dff31b5751105ae5c9be4e83fc9ab0.